src/ucp/core/ucp_request.c

```diff
@@ -396,59 +396,33 @@ static ucp_md_map_t ucp_request_get_invalidation_map(ucp_ep_h ep)
     return inv_map;
 }
 
-static void ucp_request_memh_invalidate(ucp_request_t *req, ucp_mem_h memh,
-                                        ucs_status_t status)
+int ucp_request_memh_invalidate(ucp_request_t *req, ucp_mem_h memh,
+                                ucs_status_t status)
 {
-    ucp_ep_h ep           = req->send.ep;
-    ucp_worker_h worker   = ep->worker;
-    ucp_context_h context = worker->context;
+    ucp_ep_h ep                      = req->send.ep;
+    ucp_err_handling_mode_t err_mode = ucp_ep_config(ep)->key.err_mode;
+    ucp_worker_h worker              = ep->worker;
+    ucp_context_h context            = worker->context;
     ucp_md_map_t invalidate_map;
 
+    if ((memh == NULL) || (err_mode != UCP_ERR_HANDLING_MODE_PEER) ||
+        ucp_memh_is_user_memh(memh)) {
+        ucp_request_complete_send(req, status);
```

Comment:
seems ucp_request_complete_send is called twice

Reply:
Good catch, thanks, removed!